syncer/dispatcher.go

```diff
@@ -11,9 +11,11 @@ import (
 
 var log = logging.Logger("sync.dispatch")
 
-// This is the size of the channel buffer used for receiving targets from
-// producers.
-const incomingBufferSize = 5
+// IncomingBufferSize is the size of the channel used for receiving targets from producers.
+const IncomingBufferSize = 5
+
+// WorkQueueBound is the size of the work queue
+const WorkQueueBound = 20
```

Comment:
I exposed these for tests, if its a problem I can add as constructor inputs or introduce assumptions into the tests that need a small amount of manual intervention.